### DISCUSSION TITLE
BadCredentialsException instead of DisabledException

### DISCUSSION THREAD

**Discussion by mitrae** [symfony/symfony#42656] [Q&A]
**Symfony version(s) affected**: 3.4.49

When logging as a disabled user system throws "Invalid credentials". Before(3.4.45) it was a "Blocked account" message.

Perhaps it was affected by this patch: https://symfony.com/blog/cve-2021-21424-prevent-user-enumeration-in-authentication-mechanisms

Disabled and not found customers are the same cases. Yes, I do not want to allow enumeration but how to tell the disabled customer that he is blocked and he needs to contact admin? Disabling "hide_user_not_found" is not an option.

**Comment by chalasr** [comment]
Maybe notify your customers on account status change, e.g. via email?

**Comment by derrabus** [comment]
This is not a bug and even if it was: We do not provide bugfixes for Symfony 3.4 anymore. Moving to discussions.

**Comment by S-Tian86** [comment]
Maybe you can add some help text when that error appears, something like:

Error: Invalid credentials
Message: Please contact the Administrator or reset your password.

Then if the user is disabled, when sending the email to reset password you can validate this status and notify them by email.

It is an idea, greetings.
